sync/sync_head.go

```diff
@@ -24,7 +24,7 @@ func (s *Syncer[H]) Head(ctx context.Context) (H, error) {
 	if isRecent(sbjHead, s.Params.blockTime) {
 		return sbjHead, nil
 	}
-	// otherwise, request head from a trusted peer, as we assume it is fully synced
+	// otherwise, request head from the network
```

Comment:
If the header is not recent, we request it from the network, but we **must** verify it against our current `sbjHead`, so the option should allow passing the header.

I propose defining the `With(Subjective)Head` option and making this option the flag for Exchange instructing it to request from network/random peers, while by default, it remains to ask trusted peers.

This way, we solve the problem and always verify the headers + we don't break existing default behavior.